tempesta_fw/http.c

```diff
@@ -2159,7 +2241,9 @@ tfw_http_req_process(TfwConn *conn, struct sk_buff *skb, unsigned int off)
 
 		/* Assign the right Vhost for this request. */
 		if (tfw_http_req_set_context(req)) {
-			tfw_http_conn_msg_free((TfwHttpMsg *)req);
+			TFW_INC_STAT_BH(clnt.msgs_otherr);
+			tfw_http_drop(req, 500, "cannot find"
+				      "Vhost for request", true);
```

Comment:
It's been stated that special drop/block processing is for the cases where a request either malformed or malicious (i.e. filtered out). This particular case is more of an internal error nature (or maybe even a configuration). So why `tfw_http_drop()`? Why the connection has to be closed in this case? Does the client need to know of this error in the browser?

There are still regular `tfw_http_send_500()` and `tfw_http_send_502()` that weren't converted to this block/drop style, and that's justifiable and there's no need to block a request or drop a connection in those case.

Reply:
> Does the client need to know of this error in the browser?
But in the opposite case - if we don't send error response - the client will see only hang pages.
Maybe it is more suitable for system administrator responsibility - to decide is it worth to send error responses in these cases.